src/main/java/com/acme/edu/LoggerController.java

```diff
@@ -0,0 +1,56 @@
+package com.acme.edu;
+
+import com.acme.edu.command.Command;
+import com.acme.edu.saver.Saver;
+
+import java.util.Map;
+import java.util.HashMap;
+
+public class LoggerController {
+    private static LoggerController theOnlyLogger = new LoggerController();
+    private Map<String, Boolean> commandParameters = new HashMap<>();
```

Comment:
Also you have to make Controller stateful. It will be costly to make this code parallel.